validator/tests/test_merkle_trie/tests.py

```diff
@@ -140,6 +140,77 @@ def test_merkle_trie_update(self):
             with self.assertRaises(KeyError):
                 self.get(address, ishash=True)
 
+    def test_merkle_trie_update_same_address_space(self):
```

Comment:
Could you add a doc string describing the steps of the test?  As we're currently porting this to rust, it would be good to know what the test intent is

Reply:
I have added some comments to help make more legible.